test/models/customer_test.rb

```diff
@@ -4,32 +4,35 @@
 
 class CustomerTest < ActiveSupport::TestCase
   test 'Control' do
-    customer = Customer.new
-    customer.name = 'Linda Smith'
-    customer.phone = '07594548796'
-    customer.email = 'lindasmith@gmail.com'
-    customer.address = '3 Lindow Road'
-    customer.postcode = 'SK93ME'
+    customer = create_test_customer
     assert customer.save
   end
   test 'Customer should not be saved empty' do
     customer = Customer.new
     assert_not customer.save
   end
   test 'Customer name should not contain a symbol' do
-    customer = Customer.new
+    customer = create_test_customer
     customer.name = 'R@£$'
     assert_not customer.save
   end
   test 'Customer name should allow hyphens and spaces' do
-    customer = Customer.new
+    customer = create_test_customer
     customer.name = 'Robin Haigh-Gannon'
     assert customer.save
   end
   test 'Customer phone number should not contain letters' do
-    customer = Customer.new
-    customer.name = 'Test'
+    customer = create_test_customer
     customer.phone = '0823746T673Y'
     assert_not customer.save
   end
+  def create_test_customer
```

Comment:
much cleaner!

now for some Ruby magic....

Methods in ruby can have keyword arguments and can also have default values. That means you could have a method like below:
```ruby
def create_test_customer(name: 'Test data', phone: '123456', email: 'me@text.com'...)
  customer = Customer.new
  customer.name = name
  customer.phone = phone
  ...
```
Then you can call that method passing in just the thing(s) you want to override:
```ruby
customer_with_nil_name = create_test_customer(name: nil)
```
and everything else will be set as normal.


And some Rails magic....

When you create a new instance of a model, you can pass in the attributes:
```ruby
def create_test_customer
  Customer.new(
    name: name,
    phone: phone,
    ...
  )
end
```
This is simpler and more efficient than as you have it because you are initialising the object and setting its attributes in just one call instead of 1+1 per attribute.

Reply:
Ok I have done this.